roundabout-core/src/test/java/at/fhv/itm3/s2/roundabout/util/ConfigParserTest.java

```diff
@@ -117,13 +121,95 @@ public void configParserTest_generateModel() throws ConfigParserException {
+                if (section.toString().contains("exit")) {
+                    exitSectionId = extractSectionId(section.toString());
+                } else if (section.toString().contains("track")) {
+                    Integer id = extractSectionId(section.toString());
+                    if (nextTrackSectionId != null && nextTrackSectionId != id) {
+                        assertTrue("tracks from different sections on next connector", false);
+                    }
+                    nextTrackSectionId = id;
+                } else {
+                    assertTrue("only tracks and exit on next connector", false);
+                }
+            }
+
+            Integer entrySectionId = null;
+            Integer previousTrackSectionId = null;
+            for (IProducer section : connector.getPreviousSections()) {
+                if (section.toString().contains("entry")) {
+                    Integer id = extractSectionId(section.toString());
+                    if (entrySectionId != null && entrySectionId != id) {
+                        assertTrue("entries from different sections on next connector", false);
+                    }
+                    entrySectionId = id;
+                } else if (section.toString().contains("track")) {
+                    Integer id = extractSectionId(section.toString());
+                    if (previousTrackSectionId != null && previousTrackSectionId != id) {
+                        assertTrue("tracks from different sections on next connector", false);
+                    }
+                    previousTrackSectionId = id;
+                } else {
+                    assertTrue("only tracks and entries on previous connector", false);
+                }
+            }
+
+            assertTrue("only tracks from neighbour sections", areNeighbourSections(previousTrackSectionId, nextTrackSectionId, roundaboutStructure.getStreetConnectors().size()));
+            if (exitSectionId != null) { // TODO fix after connector implementation
+                assertEquals("exit and previous tracks from same section", exitSectionId, previousTrackSectionId);
+            }
+            if (entrySectionId != null) { // TODO fix after connector implementation
+                assertEquals("entries and next tracks from same section", entrySectionId, nextTrackSectionId);
+            }
+        }
+    }
+
+    private Integer extractSectionId(String value) {
+        String regex = "(section with id )(\\d)(#\\d)";
+
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(value);
+
+        if (matcher.find()) {
+            return Integer.parseInt(matcher.group(2));
+        }
+        throw new IllegalArgumentException("id is not an integer: " + value);
+    }
+
+    private boolean areNeighbourSections(Integer previousTrackSectionId, Integer nextTrackSectionId, Integer totalSections) {
+        if (previousTrackSectionId == totalSections && nextTrackSectionId == 1) {
```

Comment:
from where does the "nextTrackSectionId == 1" come? why is it exactly 1 and not 0 or something else?

Reply:
because the test xml is defined like that